engine.py

```diff
@@ -190,24 +190,35 @@ def _get_dialog_parent(self):
 
     def _hide_menu_in_scripts(self):
         """
-        Remove the entry in the VRED Scripts window
+        Remove the entry in the VRED Scripts menu
+        If running in VRED Design also remove the Scripts menu itself
         """
         main_window = self._get_dialog_parent()
-        menu_names = main_window.menuBar().actions()
-
-        # Make it invisible for both Pro & Design
-        for menu_name in menu_names:
-            if menu_name.text() == "Scripts":
-                scripts_menu_items = menu_name.menu().actions()
-                for scripts_menu_item in scripts_menu_items:
-                    if scripts_menu_item.text() == "Shotgun":
-                        scripts_menu_item.setVisible(False)
-
-        # Check if it is Design and also "mute" the menu we made
-        if os.getenv("TK_VRED_EXECPATH").endswith("VREDDesign.exe"):
-            for menu_name in menu_names:
-                if menu_name.text() == "Scripts":
-                    menu_name.setVisible(False)
+        menu_actions = main_window.menuBar().actions()
+        scripts_action = next(
+            (
+                menu_action
+                for menu_action in menu_actions
+                if menu_action.text() == "Scripts"
+            ),
+            None,
+        )
+        shotgun_action = next(
+            (
+                submenu_action
+                for submenu_action in scripts_action.menu().actions()
+                if submenu_action.text() == "Shotgun"
+            ),
+            None,
+        )
```

Comment:
This assumes that we will always find a `"Scripts"` menu and will error if not found.

If we want to handle this error, we should add a check to see if we found the scripts menu before trying to find the shotgun menu.

Reply:
Thanks for the thoroughness on this one @staceyoue ! It is much better because of it. 👍🏻